recipes/quickstart/agents/Agents_101/Tool_Calling_101.ipynb

```diff
@@ -0,0 +1,980 @@
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "function_definitions = \"\"\"[\n",
+    "    {\n",
+    "        \"name\": \"get_user_info\",\n",
+    "        \"description\": \"Retrieve details for a specific user by their unique identifier. Note that the provided function is in Python 3 syntax.\",\n",
+    "        \"parameters\": {\n",
+    "            \"type\": \"dict\",\n",
+    "            \"required\": [\n",
+    "                \"user_id\"\n",
+    "            ],\n",
+    "            \"properties\": {\n",
+    "                \"user_id\": {\n",
+    "                \"type\": \"integer\",\n",
+    "                \"description\": \"The unique identifier of the user. It is used to fetch the specific user details from the database.\"\n",
+    "            },\n",
+    "            \"special\": {\n",
+    "                \"type\": \"string\",\n",
+    "                \"description\": \"Any special information or parameters that need to be considered while fetching user details.\",\n",
+    "                \"default\": \"none\"\n",
+    "                }\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    "]\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "system_prompt = \"\"\"You are an expert in composing functions. You are given a question and a set of possible functions. \n",
+    "Based on the question, you will need to make one or more function/tool calls to achieve the purpose. \n",
+    "If none of the function can be used, point it out. If the given question lacks the parameters required by the function,\n",
+    "also point it out. You should only return the function call in tools call sections.\n",
+    "\n",
+    "If you decide to invoke any of the function(s), you MUST put it in the format of [func_name1(params_name1=params_value1, params_name2=params_value2...), func_name2(params)]\\n\n",
+    "You SHOULD NOT include any other text in the response.\n",
+    "\n",
+    "Here is a list of functions in JSON format that you can invoke.\\n\\n{functions}\\n\"\"\".format(functions=function_definitions)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chat_history = []\n",
+    "\n",
+    "def model_chat(user_input: str, sys_prompt = system_prompt, temperature: int = 0.7, max_tokens=2048):\n",
+    "    \n",
+    "    chat_history = [\n",
+    "        {\n",
+    "            \"role\": \"system\",\n",
+    "            \"content\": system_prompt\n",
+    "        }\n",
+    "    ]\n",
+    "    \n",
+    "    chat_history.append({\"role\": \"user\", \"content\": user_input})\n",
+    "    \n",
+    "    #print(chat_history)\n",
+    "    \n",
+    "    #print(\"User: \", user_input)\n",
+    "    \n",
+    "    response = client.chat.completions.create(model=\"llama-3.2-3b-preview\",\n",
```

Comment:
we changing the model to 8B, any specific reason/ if we should hint it out for users?

Reply:
Yes-This section is for the 3.2 models-so showcasing the change in prompts for the new ones